### ISSUE TITLE
Jetty Spans disappear / are associated to a wrong trace 

### ISSUE THREAD

**Issue by josecoelho96** [open-telemetry/opentelemetry-java-instrumentation#8788]
**Describe the bug**
We are starting to collect traces from our java application, currently only relying on traces/spans collected via the supported libraries/servers, namely jetty and jdbc. After some time having traces being collected in the production environment, we noticed that some of the http requests did not appear on Grafana Tempo (the tool being used to store/view traces), but the endpoint was being called multiple times, and the response was correctly received.

This seems to be related to https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8415

The OpenTelemetry Collector metrics don't show any dropped/refused data and sampling is not active.

After some time exploring other traces, some of them were strange. Some traces had multiple child spans that weren't related to the parent span (the parent span was a http request, and the child spans were queries made by calls to other endpoints).

 This is the representation I get from a "good" trace.

![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/10677386/7fbd2fdd-d931-4fee-9cb3-55a826c9c843)

And this is the summary I have on the traces that are aggregating multiple spans.
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/10677386/c7dce3af-13df-4328-b798-454791081ba8)

Notice the duration of 50 minutes! The parent span is a normal http request. All child spans are queries that were executed because of other endpoints, so I did not expect them in the same trace. I checked the logs and the duration of the request related was average (few milliseconds)

**Steps to reproduce**
Just running the application. We have multiple instances running, with different configurations that drastically change the behaviour of some components and the problem happens on all of them. 

**What did you expect to see?**
All of the requests as a separate trace (with one or multiple child spans).

**What did you see instead?**
Only some of the traces as expected, some traces with several minutes and hundreds/thousands of child spans that have nothing to do with each other (following the code, it's impossible have the child spans related to the parent span).

**What version are you using?**
Currently 1.27.0, but also had the same symptom on 1.26.0

**Environment**
Compiler: OpenJDK 11.0.9.1
OS: Debian 11

Runtime (if different from JDK above): OpenJDK Runtime Environment (build 11.0.14+9-post-Debian-1deb11u1)
OS (if different from OS compiled on): Debian 11

**Additional context**
It's impossible to provide a sample app for the time being.

**Comment by laurit** [comment]
@josecoelho96 have you tried creating a minimal application that reproduces this issue? Can you reproduce this at will? What is the jetty version that you are using?
Probably the issue is that context from a previous request leaks over. Before the `SERVER` span is created there is a check whether current context already has a `SERVER` span. If so new `SERVER` span is not created and existing trace is continued. Try running your application `-Dotel.instrumentation.experimental.span-suppression-strategy=none` and `-Dotel.javaagent.experimental.thread-propagation-debugger.enabled=true` (or `-Dotel.javaagent.debug=true`). The first flag disables the check that prevents creating new `SERVER` span when there already is one, so you might observer your original problem going away. This flag is needed for the second flag to take effect. The second flag will trigger logging warnings from https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d38a00633accd60c8241cd7686985b5f46d6ba46/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/ContextPropagationDebug.java#L128 when a context leak is detected. Hopefully the content of these warnings will help us understand how the leak happens.

**Comment by josecoelho96** [comment]
@laurit I'm unable to create a minimal application, but I'm completely available to help testing or debugging as needed. The jetty version is 9.4.51

I was trying to reproduce the problem and I can do it. While doing some more tests, I noticed something (might just be a coincidence) - all traces were showing up correctly until I made a call to an endpoint that had to put/retrieve a file from S3 (still testing if this is true or not).

I will run the application with those flags, will update with some logs/results soon.
